src/metabase/related.clj

```diff
@@ -93,39 +93,44 @@
     (concat best (->> rest shuffle (take max-serendipity-matches)))))
 
 (def ^:private ^{:arglists '([entities])} filter-visible
-  (partial filter (fn [{:keys [archived visibility_type] :as entity}]
+  (partial filter (fn [{:keys [archived visibility_type active] :as entity}]
                     (and (or (nil? visibility_type)
-                             (= (name visibility_type) "normal"))
+                             (= (qp.util/normalize-token visibility_type) :normal))
                          (not archived)
+                         (not= active false)
```

Comment:
This is a bit confusing to me. Is this function used for metrics/segment/pulses (`archived`) and tables (`active`)? This logic doesn't seem right.

Reply:
It's used for both fields & tables (which have active) and metrics/segments (which have archived). The slightly awkward not= is there so that nil is fine as well and therefore it works for entities which do not have the active field.